sdk/clientcore/core/src/main/java/io/clientcore/core/implementation/instrumentation/fallback/FallbackInstrumentation.java

```diff
@@ -94,4 +97,36 @@ public <T> InstrumentationContext createInstrumentationContext(T context) {
             return FallbackSpanContext.INVALID;
         }
     }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean shouldInstrument(SpanKind spanKind, InstrumentationContext context) {
+        if (!isTracingEnabled) {
+            return false;
+        }
+
+        if (allowNestedSpans) {
+            return true;
+        }
+
+        return spanKind != tryGetSpanKind(context);
```

Comment:
Hmm, should `tryGetSpanKind` returning null mean instrumentation should happen?

Reply:
replied in the other comment.